api/tp/lib/src/tp/tp.c

```diff
@@ -511,6 +511,10 @@ tp_process( tp_t * ctx )
         ctx->gesture_prev         = TP_EVENT_GEST_NONE;
         ctx->release              = TP_RELEASE_DET;
         ctx->touch_prev.n_touches = 0;
+        // For TSC2003 display there can be only 1 touch, so if display is not pressed, there are no touches.
+        #ifdef __TP_TSC2003__
+        ctx->touch.n_touches = 0;
```

Comment:
How does this affect capacitive displays? If at all.

Reply:
It affects only old Resistive Displays - their touch detection approach is different from TSC2003 or FT5XX6 as STMPE protocol clears the touch detections on HW level and we handled this detection via SW. So if we clear the touch detection info here, the touch doesn't work as expected (for example, you can't hold the button pressed).

As for Capacitive displays - you are correct, by the time we get here, this variable value is already 0 as we release all the pens from the Display.